salt/cloud/__init__.py

```diff
@@ -240,7 +240,7 @@ def _opts_defaults(self, **kwargs):
                          if a.get('provider', '')]
             if providers:
                 _providers = opts.get('providers', {})
-                for provider in list(_providers).copy():
+                for provider in list(_providers)[:]:
```

Comment:
can you get rid of the list() and just make it .copy() on `_providers`

Reply:
Since _providers is a dict then yes .copy() would work on both python 2.7 and 3+ and also iterating on a dict would give the same behaviour as list(_providers) so thats good